source/D2Common/src/D2StatList.cpp

```diff
@@ -44,7 +44,8 @@ BOOL __stdcall STATLIST_AreUnitsAligned(D2UnitStrc* pUnit1, D2UnitStrc* pUnit2)
 template<class T>
 static int __fastcall StatArray_FindInsertionIndex(const T* pStatsArray, int nLayer_StatId, bool* pAlreadyInArray)
 {
-	*pAlreadyInArray = false;
+	if(pAlreadyInArray)
+		*pAlreadyInArray = false;
```

Comment:
Typically, I prefer to have braces around every code block and a space after `if`, but that's a matter of taster, I think